proto/substrait/extended_expression.proto

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: Apache-2.0
+syntax = "proto3";
+
+package substrait;
+
+import "substrait/algebra.proto";
+import "substrait/extensions/extensions.proto";
+import "substrait/plan.proto";
+import "substrait/type.proto";
+
+option csharp_namespace = "Substrait.Protobuf";
+option go_package = "github.com/substrait-io/substrait-go/proto";
+option java_multiple_files = true;
+option java_package = "io.substrait.proto";
+
+message ExpressionReference {
+  oneof expr_type {
+    Expression expression = 1;
+    AggregateFunction measure = 2;
+  }
```

Comment:
This seems a little odd.  Perhaps we should have `ScalarExpression` and `AggregateExpression`?  Would it be possible to have an aggregate expression like `SUM(x) * 3`?

Reply:
I agree.  I don't know of any significant examples of "aggregate expressions" and "aggregate function followed by scalar expression" should be sufficient..